### PR TITLE
Rearrange origination fees

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

File: contracts/truefi/TrueLender.sol [hunk 1]
```diff
@@ -77,6 +77,9 @@ contract TrueLender is ITrueLender, Ownable {
     // maximum amount of loans lender can handle at once
     uint256 public maxLoans;
 
+    // implemented as an ERC20, will change after implementing stkPool
+    IERC20 public stakePool;
```
Comment (barrutko) [inline contracts/truefi/TrueLender.sol:81]: `stakePool ` => `stakingPool `

File: test/truefi/TrueLender.test.ts [hunk 2]
```diff
@@ -343,13 +374,20 @@ describe('TrueLender', () => {
         expect('fund').to.be.calledOnContractWith(mockLoanToken, [])
       })
 
+      it('transfers origination fee to stake pool', async () => {
```
Comment (vanruch) [inline test/truefi/TrueLender.test.ts:377]: origination?

**Review by vanruch — APPROVED** [review]
Looks good. I only don't like the name `origination fees`. What does this even mean in this context?

**Comment by MiksuJak** [comment]
> Looks good. I only don't like the name `origination fees`. What does this even mean in this context?

https://github.com/trusttoken/truefi-spec/blob/v2/truefi-v2.md
got it from specs

File: contracts/truefi/TrueLender.sol (hunk 1 above)
Comment (MiksuJak) [reply contracts/truefi/TrueLender.sol]: 🤝

**Review by vanruch — CHANGES_REQUESTED** [review]
We need to convert it to LP tokens

File: contracts/truefi/TrueFiPool.sol [hunk 3]
```diff
@@ -153,7 +154,7 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
      * @dev only lender can perform borrowing or repaying
      */
     modifier onlyLender() {
-        require(msg.sender == address(_lender), "TrueFiPool: Only lender can borrow or repay");
+        require(msg.sender == address(_lender), "TrueFiPool: Caller is not the lender");
```
Comment (barrutko) [inline contracts/truefi/TrueFiPool.sol:157]: 👍

**Review by barrutko — APPROVED** [review]
Imo looks fine, but still would be nice if @hal909 and @vanruch had a look

File: contracts/truefi/TrueFiPool.sol [hunk 4]
```diff
@@ -258,15 +258,7 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
...
-        uint256 amountToDeposit = amount.sub(fee);
-        uint256 amountToMint = amountToDeposit;
-
-        // first staker mints same amount deposited
-        if (totalSupply() > 0) {
-            amountToMint = totalSupply().mul(amountToDeposit).div(poolValue());
-        }
-        // mint pool tokens
-        _mint(msg.sender, amountToMint);
+        uint256 amountToMint = mint(amount.sub(fee));
```
Comment (vanruch) [inline contracts/truefi/TrueFiPool.sol:261]:
```suggestion
        uint256 mintedAmount = mint(amount.sub(fee));
```

File: contracts/truefi/TrueFiPool.sol [hunk 5]
```diff
@@ -542,4 +530,21 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
     function currencyBalance() internal view returns (uint256) {
         return _currencyToken.balanceOf(address(this)).sub(claimableFees);
     }
+
+    function mint(uint256 depositedAmount) internal returns (uint256) {
+        uint256 amountToMint = depositedAmount;
+
+        if (amountToMint == 0) {
```
Comment (vanruch) [inline contracts/truefi/TrueFiPool.sol:537]: Maybe move this if one line up
`if (depositedAmount == 0) { ... `

File: contracts/truefi/mocks/MockStakingPool.sol [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {ERC20} from "./../common/UpgradeableERC20.sol";
+import {ITrueFiPool} from "./../interface/ITrueFiPool.sol";
+
+contract MockStakingPool is ERC20 {
+    ITrueFiPool pool;
+
+    function setPool(ITrueFiPool _pool) public {
```
Comment (vanruch) [inline contracts/truefi/mocks/MockStakingPool.sol:10]: why not with constructor?

File: contracts/truefi/TrueLender.sol [hunk 7]
```diff
@@ -77,6 +77,9 @@ contract TrueLender is ITrueLender, Ownable {
     // maximum amount of loans lender can handle at once
     uint256 public maxLoans;
 
+    // implemented as an ERC20, will change after implementing stkPool
+    IERC20 public stakingPool;
```
Comment (vanruch) [inline contracts/truefi/TrueLender.sol:81]: Is there a reason, we can't keep it as a simple address?

Comment (MiksuJak) [reply contracts/truefi/TrueLender.sol]: You're right, thank You for that :)

**Review by vanruch — APPROVED** [review]

**Review by hal909 — APPROVED** [review]

**Pull request merged**